### ISSUE TITLE
Non-conformant behavior regarding multiple function declarations with default arguments

### ISSUE THREAD

**Issue by Endilll** [llvm/llvm-project#59363]
[[over.match.best]/4](https://eel.is/c++draft/over.match.best#general-4) includes an example with ill-formed code that Clang fails to diagnose:
```cpp
namespace A {
  extern "C" void f(int = 5);
}
namespace B {
  extern "C" void f(int = 5);
}

using A::f;
using B::f;

void use() {
  f(3);             // OK, default argument was not used for viability
  f();              // error: found default argument twice
}
```
For the reference, [[dcl.fct.default]/4](https://eel.is/c++draft/dcl.fct.default#4) states that "declarations that inhabit different scopes have completely distinct sets of default arguments."
There's also a related [CWG418](https://wg21.link/cwg418) and a [D139429](https://reviews.llvm.org/D139429) that adds a test for it.

**Comment by shafik** [comment]
Confirmed: https://godbolt.org/z/sTqPqdeWr

Note gcc also does not catch this but MSVC does.

**Comment by cor3ntin** [comment]
There is already a bunch of FIXME there
https://github.com/llvm/llvm-project/blob/cfd44221e3e1783c0f44d5b1694dfbe84187246a/clang/lib/Sema/SemaLookup.cpp#L395-L407

**Comment by zygoloid** [comment]
This is CWG1.
